docs/adr/001_error_handling.md

```diff
@@ -0,0 +1,107 @@
+# Error handling patterns in public API interfaces
+
+## Date
+17 Feb 2025 
+
+## Accepted Option
+
+**Option 3** 
+
+## Context and Problem Statement
+There is uncertainty around how to model errors in the `opentelemetry-rust` public API interfaces - that is, APIs that are exposed to users of the project's published crates. This is for example the case with the exporter traits - [SpanExporter](https://github.com/open-telemetry/opentelemetry-rust/blob/eca1ce87084c39667061281e662d5edb9a002882/opentelemetry-sdk/src/trace/export.rs#L18), [LogExporter](https://github.com/open-telemetry/opentelemetry-rust/blob/eca1ce87084c39667061281e662d5edb9a002882/opentelemetry-sdk/src/logs/export.rs#L115), and [PushMetricExporter](https://github.com/open-telemetry/opentelemetry-rust/blob/eca1ce87084c39667061281e662d5edb9a002882/opentelemetry-sdk/src/metrics/exporter.rs#L11) which form part of the API surface of `opentelemetry-sdk`.
+
+We will focus on the exporter traits in this example, but the outcome should be applied to _all_ public traits and their fallible operations. 
+
+There are various ways to handle errors on trait methods, including swallowing them and logging, panicing, returning a shared global error, or returning a method-specific error. We strive for consistency, and we want to be sure that we've put enough thought into what this looks like that we don't have to make breaking interface changes unecessarily in the future.
+
+This was discussed extensively in [#2571](https://github.com/open-telemetry/opentelemetry-rust/issues/2571).
+
+
+## Related Work
+
+* [#2564](https://github.com/open-telemetry/opentelemetry-rust/issues/2564)
+* [#2561](https://github.com/open-telemetry/opentelemetry-rust/issues/2561)
+* [#2381](https://github.com/open-telemetry/opentelemetry-rust/issues/2381)
+
+## Considered Options
+
+**Option 1: Continue as is**
```

Comment:
I still suggest to only list the decision here to avoid long reads. At the end of the doc, we can mention considered-alternatives, and move this there.